### ISSUE TITLE
Indenting elements for formatting ps1xml files don't work

### ISSUE THREAD

**Issue by DJackman123** [PowerShell/PowerShell#16960]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

The [reference docs for formatting files](https://docs.microsoft.com/en-us/powershell/scripting/developer/format/frame-element-for-customitem-for-controls-for-configuration-format?view=powershell-7.2) say that the Frame element inside of CustomItem can contain elements to specify indenting that will happen to the formatted text: LeftIndent, RightIndent, FirstLineIndent, and/or FirstLineHanging.  However, none of these elements actually do any indenting to the output.
From what I can tell, Windows PowerShell 5.1 has the same problem.

The following is an example formatting ps1xml file and script that demonstrates the problem:

FormatBug.ps1xml:
```
<?xml version="1.0" encoding="utf-8"?>
<Configuration>
  <Controls>
    <Control>
      <Name>LongMessageInfo-Grouping</Name>
      <CustomControl>
        <CustomEntries>
          <CustomEntry>
            <CustomItem>
              <Frame>
                <LeftIndent>10</LeftIndent>
                <RightIndent>10</RightIndent>
                <FirstLineHanging>8</FirstLineHanging>
                <!-- <FirstLineIndent>3</FirstLineIndent> -->
                <CustomItem>
                  <NewLine />
                  <Text>Start Group: </Text>
                  <ExpressionBinding>
                    <!-- <PropertyName>Start</PropertyName> -->
                    <ScriptBlock>"[$($_.Start)] $((@($_.Message) * $_.RepeatCount) -join ' ')"</ScriptBlock>
                  </ExpressionBinding>
                </CustomItem>
              </Frame>
            </CustomItem>
          </CustomEntry>
        </CustomEntries>
      </CustomControl>
    </Control>
  </Controls>
  <ViewDefinitions>
    <View>
      <Name>LongMessageInfo</Name>
      <ViewSelectedBy>
        <TypeName>LongMessageInfo</TypeName>
      </ViewSelectedBy>
      <GroupBy>
        <PropertyName>Start</PropertyName>
        <CustomControlName>LongMessageInfo-Grouping</CustomControlName>
      </GroupBy>
      <CustomControl>
        <CustomEntries>
          <CustomEntry>
            <CustomItem>
              <Frame>
                <LeftIndent>3</LeftIndent>
                <FirstLineHanging>4</FirstLineHanging>
                <!-- <FirstLineIndent>2</FirstLineIndent> -->
                <CustomItem>
                  <ExpressionBinding>
                    <ScriptBlock>"$($_.Start) $((@($_.Message) * $_.RepeatCount) -join ' ')"</ScriptBlock>
                  </ExpressionBinding>
                </CustomItem>
              </Frame>
            </CustomItem>
          </CustomEntry>
        </CustomEntries>
      </CustomControl>
    </View>
  </ViewDefinitions>
</Configuration>
```

FormatBug.ps1:
```
class LongMessageInfo
{
    [string]$Start
    [string]$Message
    [int]$RepeatCount
}

Update-FormatData '.\FormatBug.ps1xml'

$longMessage = [LongMessageInfo]::new()
$longMessage.Start = 'start'
$longMessage.Message = 'This is a long message'
$longMessage.RepeatCount = 30

Write-Output $longMessage
```

### Expected behavior

```console
Output text is indented according to the formatting file (left, right, and first-line indents).
```

### Actual behavior

```console
Output text has no indents at all.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
> Update-FormatData

Try `Update-FormatData -PrependPath file.ps1xml`

**Comment by 237dmitry** [comment]
Try `-AppendPath` parameter.

```console
Use this parameter to add a file that formats a .NET object that is not referenced in the built-in formatting files.
```

**Comment by DJackman123** [comment]
> Try `-AppendPath` parameter.

`-AppendPath` is the position 0 parameter.  So `Update-FormatData .\FormatBug.ps1xml` is the same as `Update-FormatData -AppendPath .\FormatBug.ps1xml`.  And, before you say anything, `-PrependPath` doesn't make any different here, either.

**Comment by jhoneill** [comment]
Working XML
```
<?xml version="1.0" encoding="utf-8"?>
<Configuration>
    <Controls>
        <Control>
            <Name>LongMessageInfo-Grouping</Name>
            <CustomControl>
                <CustomEntries>
                    <CustomEntry>
                        <CustomItem>
                            <Frame>
                                <LeftIndent>10</LeftIndent>
                                <RightIndent>10</RightIndent>
                                <FirstLineHanging>8</FirstLineHanging>
                                <CustomItem>
                                    <Text>Start group </Text>
                                    <ExpressionBinding>
                                        <ScriptBlock>
                                            <ScriptBlock>"[$($_.Start)] $((@($_.Message) * $_.RepeatCount) -join ' ')"</ScriptBlock>
                                        </ScriptBlock>
                                    </ExpressionBinding>
                                            <NewLine/>
                                </CustomItem>
                            </Frame>
                        </CustomItem>
                    </CustomEntry>
                </CustomEntries>
            </CustomControl>
        </Control>
    </Controls>
    <ViewDefinitions>
        <View>
            <Name>LongMessageInfo</Name>
            <ViewSelectedBy>
            <TypeName>LongMessageInfo</TypeName>
            </ViewSelectedBy>
            <GroupBy>
                <PropertyName>Start</PropertyName>
                <CustomControlName>LongMessageInfo-Grouping</CustomControlName>
            </GroupBy>
            <CustomControl>
                <CustomEntries>
                    <CustomEntry>
                        <CustomItem>
                            <Frame>
                                <LeftIndent>3</LeftIndent>
                                <FirstLineHanging>4</FirstLineHanging>
                                <!-- <FirstLineIndent>2</FirstLineIndent> -->
                                <CustomItem>
                                    <ExpressionBinding>
                                        <ScriptBlock>
                                            <ScriptBlock>"$($_.Start) $((@($_.Message) * $_.RepeatCount) -join ' ')"</ScriptBlock>
                                        </ScriptBlock>
                                    </ExpressionBinding>
                                    <NewLine/>                    
                                </CustomItem>
                            </Frame>
                        </CustomItem>
                    </CustomEntry>
                </CustomEntries>
            </CustomControl>
        </View>
    </ViewDefinitions>
</Configuration>
```

**Comment by DJackman123** [comment]
Thanks for providing this.  Based on this I played around with it a bit and determined that it only works when the ExpressionBinding is followed by a NewLine.  The docs say nothing about this.  But is it a doc bug or a bug in the implementation?  Should the NewLine really be required?  I didn't want my items to be separated by linefeeds--the whole point of the hanging indent was to avoid this.

**Comment by DJackman123** [comment]
How do I re-open and un-answer this bug?  Based on the last statement this is still a bug.

**Comment by jhoneill** [comment]
The <scriptblocks> in my sample were doubled up. although that does not seem to change anything. 
This version removes them and the line break after expression binding
this works
```
<?xml version="1.0" encoding="utf-8"?>
<Configuration>
    <Controls><Control><Name>LongMessageInfo-Grouping</Name><CustomControl><CustomEntries><CustomEntry><CustomItem><Frame>
            <LeftIndent>10</LeftIndent><RightIndent>10</RightIndent><FirstLineHanging>8</FirstLineHanging>
            <CustomItem><Text>Start group</Text><ExpressionBinding><ScriptBlock>"[$($_.Start)] $((@($_.Message + "~") * $_.RepeatCount) -join ' ')"</ScriptBlock></ExpressionBinding><NewLine/></CustomItem>
    </Frame></CustomItem></CustomEntry></CustomEntries></CustomControl></Control></Controls>
    <ViewDefinitions><View><Name>LongMessageInfo</Name><ViewSelectedBy><TypeName>LongMessageInfo</TypeName></ViewSelectedBy>
        <GroupBy><PropertyName>Start</PropertyName><CustomControlName>LongMessageInfo-Grouping</CustomControlName></GroupBy>
        <CustomControl><CustomEntries><CustomEntry><CustomItem><Frame>
            <LeftIndent>3</LeftIndent><FirstLineHanging>4</FirstLineHanging>
            <!-- <FirstLineIndent>2</FirstLineIndent> -->
            <CustomItem><ExpressionBinding><ScriptBlock>"$($_.Start) $((@($_.Message+ "#") * $_.RepeatCount) -join ' ')"</ScriptBlock></ExpressionBinding><NewLine/></CustomItem>
        </Frame></CustomItem></CustomEntry></CustomEntries>        </CustomControl>
    </View></ViewDefinitions>
</Configuration>
```

(ignore the ~ and # being inserted they are so can see the code has been reloaded)

```
> Update-FormatData
> Write-Output $longMessage


          Start group[start] This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a
                  long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is
                  a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This
                  is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~
                  This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long message~ This is a long
                  message~ This is a long message~


   start This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long    
       message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a  
       long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This  
       is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message# This is a long message#  
       This is a long message# This is a long message# This is a long message#

```


Even this works
```
<?xml version="1.0" encoding="utf-8"?>
<Configuration><Controls><Control><Name>LongMessageInfo-Grouping</Name><CustomControl><CustomEntries><CustomEntry><CustomItem><Frame><LeftIndent>10</LeftIndent><RightIndent>10</RightIndent><FirstLineHanging>8</FirstLineHanging><CustomItem><Text>Start group</Text><ExpressionBinding><ScriptBlock>"[$($_.Start)] $((@($_.Message + "~") * $_.RepeatCount) -join ' ')"</ScriptBlock></ExpressionBinding><NewLine/></CustomItem></Frame></CustomItem></CustomEntry></CustomEntries></CustomControl></Control></Controls><ViewDefinitions><View><Name>LongMessageInfo</Name><ViewSelectedBy><TypeName>LongMessageInfo</TypeName></ViewSelectedBy><GroupBy><PropertyName>Start</PropertyName><CustomControlName>LongMessageInfo-Grouping</CustomControlName></GroupBy><CustomControl><CustomEntries><CustomEntry><CustomItem><Frame><LeftIndent>3</LeftIndent><FirstLineHanging>4</FirstLineHanging><CustomItem><ExpressionBinding><ScriptBlock>"$($_.Start) $((@($_.Message+ "#") * $_.RepeatCount) -join ' ')"</ScriptBlock></ExpressionBinding><NewLine/></CustomItem></Frame></CustomItem></CustomEntry></CustomEntries></CustomControl></View></ViewDefinitions></Configuration>
```

**Comment by DJackman123** [comment]
I wasn't talking about whitespace in the XML.  I was talking about the `<NewLine/>` element in the XML.  Is the `<NewLine/>` element after the `<ExpressionBinding>` element required?  The docs don't say that it is, and I really don't want to have a blank line after each of the objects just so I can get the indentation to work.

**Comment by jhoneill** [comment]
@DJackman123  Sorry that's a loud DOH ! from me.  I read it as a "Line feed" as a "<Newline />" .
 I get the same behaviour, remove the newline and the indents don't work. 

I've written quite a few format files, but I don't use the <indent> tags. I did a search through all the Ps1xml files I have and the only indents in them a left, and the sampling I did showed everyone has a newline combined with it. I can't see why this would need to be so. It may be intentional but not properly documented or it may a long standing bug.
